File: test/hazard.cc
#include "script.h"
#include "state.h"

struct Hazard : public state_t {
    void go();
};


void Hazard::go()
{
    stack[0] = NULL;
    executed = 0;

    // Put 0 into [0], 255 into [1] and then flip between them.
    XOR(A);
    STA(A);
    LOAD(X,A);
    INC(Y,X);
    DEC(A);
    STA(Y);
    LOADM(A,X);
    LOADM(A,Y);
    LOADM(A,X);
    LOADM(A,Y);
    RET();
}


int main(int argc, char * argv[])
{
    sim_main(Hazard(), argc, argv);
}

File: src/hazard.rs
use instructions::constants::*;

pub mod disassemble;
pub mod emitter;
pub mod instructions;
pub mod resistors;
pub mod spice_check;
pub mod spice_load;
pub mod state;

fn main() {
    let mut hazard = instructions::Instructions::default();

    hazard
        .xor  (A)
        .sta  (A)
        .load (X, A)
        .incv (Y, X)
        .dec  (A)
        .sta  (Y)
        .load (A, [X])
        .load (A, [Y])
        .load (A, [X])
        .load (A, [Y])

        .ret();

    spice_check::spice_check_args(|_| hazard)
}